projects/sandbox/train/train/train.py

```diff
@@ -228,92 +212,115 @@ def main(
             outdir,
-            monitor,
-            threshold=threshold,
-            additional=additional,
+            f"valid_auroc@{max_fpr:0.1e}",
             early_stop=early_stop,
             checkpoint_every=checkpoint_every,
         )
 
-        # pass this all to a validation callable which will
-        # build the necessary datasets, compute predictions
-        # on them using the model, and pass the predictions
-        # to the recorder
-        validator = Validator(
-            recorder,
-            background=valid_background,
-            glitches=valid_glitches,
-            responses=valid_responses,
-            snr_thresh=snr_thresh,
-            highpass=highpass,
-            kernel_length=kernel_length,
+        # hard code some of these validation parameters
+        # for now until it seems like they might need
+        # some tuning
+        validator = valid_utils.Validator(
+            tracker,
+            valid_background,
+            valid_waveforms,
+            sample_rate,
             stride=valid_stride,
-            sample_rate=sample_rate,
+            injection_stride=4,
+            kernel_length=kernel_length,
             batch_size=4 * batch_size,
-            glitch_frac=glitch_prob,
+            pool_length=4,
+            integration_length=1,
+            livetime=valid_livetime,
+            shift=1,
+            max_fpr=max_fpr,
             device=device,
         )
     else:
+        tf = t0 + duration
+        rescaler.fit(*background)
         validator = None
+    rescaler = rescaler.to(device)
 
-    # fit rescaler to training background
-    augmentor.rescaler.fit(*background)
-    augmentor.to(device)
+    # load our glitches and build an object
+    # for randomly sampling from them
+    logging.info("Loading glitches")
+    glitches = train_utils.get_glitches(glitch_dataset, ifos, tf)
+    for ifo, dset in glitches.items():
+        logging.info(f"{len(dset)} glitches from {ifo}")
+    glitch_sampler = GlitchSampler(
+        prob=glitch_prob,
+        max_offset=int(trigger_distance * sample_rate),
+        **glitches,
+    )
+
+    # now construct an object that will make
+    # real-time augmentations to our training
+    # data as its loaded
+    augmentor = train_utils.get_augmentor(
+        ifos,
+        sample_rate,
+        glitch_sampler,
+        waveforms,
+        waveform_prob,
+        snr_sampler,
+        rescaler,
+        mute_frac,
+        swap_frac,
+        glitch_downweight,
+        trigger_distance,
+    )
+    augmentor = augmentor.to(device)
 
-    # create full training dataloader
+    # Create full training dataloader.
+    # Use waveform dataset to dictate what
+    # an "epoch" should be, adding a factor
+    # to account for our sky parameter sampling
+    # and to balance compute vs. validation resolution
+    waveforms_per_batch = batch_size * waveform_prob
+    batches_per_epoch = 4 * len(waveforms) / waveforms_per_batch
```

Comment:
This needs to be an `int`

Reply:
Ah had this fixed in another branch, great catch